### PR TITLE
Issue 1441 - Sort all 3D Repo UI lists and add search functionality.

### LINKED ISSUE #1441: Sort All 3D Repo UI Lists and add Search Functionality.
### Description
A selection of lists on the front-end User Interface are currently not sorted in alphabetical order.

Furthermore the user does not have the ability to search for a list option, as in the list's are not searchable by keystroke. 

For example : if i type A on a job's list i would not be taken to any matching job titles such as "Architect". 

Any list's that have a logical order in place such as **Priority** "Low, Medium, High" or **Status** "Open, In Progress, For Approval, Closed" will not be altered.

### Goals
- [x] View any UI list in Alphabetical Order.
- [x] Use the keyboard to search by letter. 

### Tasks

- [x] Identify the UI list's that require sorting.
    * Check and note all list's that have a logical order.
- [x] Apply the sort to the lists. 
- [x] Add the search functionality. 
- [x] Test the search functionality. 
- [ ] Write a sorting unit test. 

### High Priority - Viewer lists. (Any list's found in the viewer)
- [x] Issues Card - Assign User list (This will be completed when, this branch and #1272 are both merged into staging as the logic is shared with Safetibase Job's List)
- [x] Issues Card - Types List  - (This has been done on #1272 )
- [x] Safetibase - Risk Owner 
- [x] Safetibase - Category 

### Medium Priority - Teamspace lists. 
- [x] Teamspace - New Model (Model Type)
- [x] Teamspace - Teamspace Selection NB: This needs to be checked if not already sorted.  

*The options below are the same lists but displayed in two different options, these still need to be checked and sorted if necessary.*
- [x] Teamspace Selection - found when clicking the large green plus icon.
- [x] Teamspace Selection - found when clicking the nested smaller plus icon on Model & Federations.

###  Low Priority - User Management
- [x] User management - Project List  

**NB** If any list contains an option such as "Unassigned" or "Other", these options must remain at the bottom of the lists.

### PR DESCRIPTION
**Pull request by niblux**
This fixes #1441

#### Description
This branches sorts all the lists below in alphabetical order, and also allows the user to search for an option using a keystroke. ie: When a user presses "a" in an Assign list , and Architect would be highlighted and pre-selected.
e.g.
 - Issues Card - Assign User list 
   - ( This will be completed when, this branch and #1272 are both merged into staging as the sorting logic is shared with Safetibase Risk Owner List).
  - Issues Card - Types List - (This has been implemented on #1272  )
  - Safetibase - Risk Owner
  - Safetibase - Category
  - User Management - Project List 
  - Teamspaces List - are already in alphabetical order , with the current username at the top of the list.
  - Model Types - When selecting both inner grey plus icon , and big green button plus icon on Teamspaces.
 
#### Test cases
- All lists mentioned above are in alphabetical order, (aside from the current username, on Teamspaces).
- All lists above should allow the user to search by keystroke.

### CODE REVIEW

**Comment by niblux** [comment]
@simplypixi , @dleszcz  Hi Guys, i have made some changes to the CellSelect styles as one of the feature requests was to use the same keypress search functionality you guys made on the "SelectField" component. So i just changed the component used from Select to SelectField. Hope this is ok :) .

Could you please take a quick look and make sure i'm on the right track with the suggested change.

File: frontend/modules/jobs/jobs.redux.ts [hunk 1]
```diff
@@ -37,7 +38,10 @@ export const INITIAL_STATE = {
 	colors: [],
 	myJob: {}
 };
-export const fetchJobsSuccess = (state = INITIAL_STATE, { jobs }) => ({ ...state, jobs });
+export const fetchJobsSuccess = (state = INITIAL_STATE, { jobs }) => {
+	jobs = sortByField([...jobs], { order: 'asc', config: { field: '_id' } });
+	return{ ...state, jobs };
```
Comment (simplypixi) [inline frontend/modules/jobs/jobs.redux.ts]:
```suggestion
	return { ...state, jobs };
```

File: frontend/routes/components/selectField/selectField.component.tsx [hunk 2]
```diff
@@ -108,7 +108,7 @@ export class SelectField extends React.PureComponent<any, any> {
 		};
 
 		return (
-			<Select className={className} {...selectProps} MenuProps={customMenuProps}>
+			<Select classes={classes} className={className} {...selectProps} MenuProps={customMenuProps}>
```
Comment (simplypixi) [inline frontend/routes/components/selectField/selectField.component.tsx]: I think you no need extra `classes` prop. It's included on `selectProps`, which is also passed to this component ;)

File: frontend/routes/components/selectField/selectField.component.tsx [hunk 3]
```diff
@@ -17,10 +17,10 @@
 
 import * as React from 'react';
 
-import Select from '@material-ui/core/Select';
+import Select, { SelectProps } from '@material-ui/core/Select';
 import { MuiTheme } from '../../../styles/theme';
 
-export class SelectField extends React.PureComponent<any, any> {
+export class SelectField extends React.PureComponent<SelectProps, any> {
```
Comment (simplypixi) [inline frontend/routes/components/selectField/selectField.component.tsx]: 👍

File: frontend/routes/projects/projects.component.tsx [hunk 4]
```diff
@@ -28,6 +28,7 @@ import { ProjectsPermissions } from '../projectsPermissions';
 import { ModelsPermissions } from '../modelsPermissions';
 import { Container, Options, SelectContainer, SwitchButton, IconLeft, IconRight } from './projects.styles';
 import { CellSelect } from '../components/customTable/components/cellSelect/cellSelect.component';
+import { sortByField } from '../../helpers/sorting';
```
Comment (simplypixi) [inline frontend/routes/projects/projects.component.tsx]: unnecessary import

File: frontend/constants/model-permissions.ts [hunk 5]
```diff
@@ -17,14 +17,9 @@ export const MODEL_ROLES_DESC = {
...
 	{
-		key: MODEL_ROLES_TYPES.UNASSIGNED,
-		label: 'Unassigned',
-		tooltip: MODEL_ROLES_DESC.UNASSIGNED
-	},
-	{
-		key: MODEL_ROLES_TYPES.VIEWER,
-		label: 'Viewer',
-		tooltip: MODEL_ROLES_DESC.VIEWER
+		key: MODEL_ROLES_TYPES.ADMINISTRATOR,
```
Comment (carmenfan) [inline frontend/constants/model-permissions.ts]: I don't think you want to shift this order.

The order of this is intention by level of permissions

Comment (niblux) [reply frontend/constants/model-permissions.ts]: Hi @carmenfan , i have put these back into their original order 👍

Comment (niblux) [reply frontend/constants/model-permissions.ts]: > image
> 
> Models in new federation tab, project in new model tab and project in user permissions tab should be sorted

I have resolved these lists carmen, it required additional sorting on the teamspaces.redux as well as the userManagement. 

Unlike the jobs.redux which only had to be done in one place. :)

**Pull request merged**